### PR TITLE
Fix HTTP 500 for ranked cancer variants

### PR DESCRIPTION
**Pull request by maehler**
This PR adds a functionality or fixes a bug.

When trying to access the variant details page for a ranked cancer variant, a HTTP 500 was returned. From the stack trace it looked like it was isolated to the rendering of the rank score panel.

I was looking into making a test case for this, but gave up when it looked like I would have to set up a parallel test track with fixtures and all in order to get that working. I'd be happy to contribute to this, but before diving into that I'd like to see that my suggested changes make sense to start with.

<details>
<summary>Stack trace</summary>

```
2022-06-30 09:12:34 vs842 scout.server.blueprints.variant.views[14764] DEBUG Variants view requesting data for variant cfc2d0bd61fad4e05fc2d5e2590aecf3
2022-06-30 09:12:34 vs842 scout.adapter.mongo.case[14764] INFO Fetching case somatic_ranking_test institute clingen
2022-06-30 09:12:34 vs842 scout.adapter.mongo.case[14764] INFO Fetching case somatic_ranking_test
2022-06-30 09:12:34 vs842 scout.adapter.mongo.hgnc[14764] DEBUG Fetching gene 6342
2022-06-30 09:12:34 vs842 scout.adapter.mongo.omim[14764] DEBUG Fetching all diseases for gene 6342
2022-06-30 09:12:34 vs842 scout.server.blueprints.variant.controllers[14764] DEBUG Variant category cancer
2022-06-30 09:12:34 vs842 scout.adapter.mongo.hgnc[14764] DEBUG Fetching gene 6342
2022-06-30 09:12:34 vs842 scout.adapter.mongo.omim[14764] DEBUG Fetching all diseases for gene 6342
2022-06-30 09:12:34 vs842 scout.server.utils[14764] WARNING VCF file does not seem to exist
2022-06-30 09:12:34 vs842 scout.adapter.mongo.event[14764] DEBUG Fetching all comments for institute clingen case somatic_ranking_test variant a76a271c34dd00584f03e8c5da097e77
2022-06-30 09:12:34 vs842 scout.adapter.mongo.institute[14764] DEBUG Fetching all institutes
2022-06-30 09:12:34 vs842 scout.server.blueprints.variant.views[14764] DEBUG Fetching loqusdb information for cfc2d0bd61fad4e05fc2d5e2590aecf3
[2022-06-30 09:12:34 +0200] [14764] [ERROR] Error handling request /clingen/somatic_ranking_test/cancer/cfc2d0bd61fad4e05fc2d5e2590aecf3?cancer=yes
Traceback (most recent call last):
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/gunicorn/workers/sync.py", line 136, in handle
    self.handle_request(listener, req, client, addr)
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/gunicorn/workers/sync.py", line 179, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/flask/app.py", line 2095, in __call__
    return self.wsgi_app(environ, start_response)
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/werkzeug/middleware/proxy_fix.py", line 187, in __call__
    return self.app(environ, start_response)
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/flask/app.py", line 2080, in wsgi_app
    response = self.handle_exception(e)
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/flask/app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/flask/app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/flask/app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/flask/app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/scout_prod/scout/scout/server/utils.py", line 90, in decorated_function
    return render_template(template_name, **context)
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/flask/templating.py", line 148, in render_template
    return _render(
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/flask/templating.py", line 128, in _render
    rv = template.render(context)
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/scout_prod/scout/scout/server/blueprints/variant/templates/variant/cancer-variant.html", line 280, in top-level template code
    {{ alignments(institute, case, variant, current_user, config, igv_tracks, splice_junctions_tracks) }}
  File "/scout_prod/scout/scout/server/templates/layout.html", line 1, in top-level template code
    {% extends "bootstrap_global.html" %}
  File "/scout_prod/scout/scout/server/templates/bootstrap_global.html", line 62, in top-level template code
    {% block body %}
  File "/scout_prod/scout/scout/server/templates/bootstrap_global.html", line 65, in block 'body'
    {% block content %}
  File "/scout_prod/scout/scout/server/templates/layout.html", line 93, in block 'content'
    {% block content_main %}{% endblock %}
  File "/scout_prod/scout/scout/server/blueprints/variant/templates/variant/cancer-variant.html", line 108, in block 'content_main'
    <div class="col-12">{{ rankscore_panel(variant) }}</div>
  File "/home/genetik/.pyenv/versions/scout/lib/python3.9/site-packages/jinja2/utils.py", line 83, in from_obj
    if hasattr(obj, "jinja_pass_arg"):
jinja2.exceptions.UndefinedError: the template 'variant/utils.html' (imported on line 3 in 'variant/cancer-variant.html') does not export the requested name 'rankscore_panel'
```

</details>

**How to test**:
1. Load a case with `vcf_cancer` set to a VCF file that has been ranked with genmod.
2. Navigate to the list of clinical variants in scout.
3. Click the link for any variant in the list.

**Expected outcome**:
The variant page should be displayed.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
Thanks for the fix @maehler. I'll be able to review tomorrow morning!

**Comment by northwestwitch** [comment]
Tested locally by converting the demo RD case into a cancer case.

I can reproduce the error:

```
2022-07-01 06:23:48 ChiaraRMBP.local urllib3.connectionpool[34125] DEBUG Starting new HTTPS connection (1): raw.githubusercontent.com:443
2022-07-01 06:23:49 ChiaraRMBP.local urllib3.connectionpool[34125] DEBUG https://raw.githubusercontent.com:443 "GET /Clinical-Genomics/reference-files/master/rare-disease/rank_model/rank_model_-v1.20-.ini HTTP/1.1" 200 2168
2022-07-01 06:23:49 ChiaraRMBP.local scout.server.app[34125] ERROR Exception on /cust000/643594/cancer/4c7d5c70d955875504db72ef8e1abe77 [GET]
Traceback (most recent call last):
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/utils.py", line 90, in decorated_function
    return render_template(template_name, **context)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/templating.py", line 148, in render_template
    return _render(
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/templating.py", line 128, in _render
    rv = template.render(context)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/jinja2/environment.py", line 1301, in render
    self.environment.handle_exception()
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/jinja2/environment.py", line 936, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/blueprints/variant/templates/variant/cancer-variant.html", line 280, in top-level template code
    {{ alignments(institute, case, variant, current_user, config, igv_tracks, splice_junctions_tracks) }}
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/templates/layout.html", line 1, in top-level template code
    {% extends "bootstrap_global.html" %}
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/templates/bootstrap_global.html", line 62, in top-level template code
    {% block body %}
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/templates/bootstrap_global.html", line 65, in block 'body'
    {% block content %}
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/templates/layout.html", line 93, in block 'content'
    {% block content_main %}{% endblock %}
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/server/blueprints/variant/templates/variant/cancer-variant.html", line 108, in block 'content_main'
    <div class="col-12">{{ rankscore_panel(variant) }}</div>
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/jinja2/utils.py", line 83, in from_obj
    if hasattr(obj, "jinja_pass_arg"):
jinja2.exceptions.UndefinedError: the template 'variant/utils.html' (imported on line 3 in 'variant/cancer-variant.html') does not export the requested name 'rankscore_panel'
2022-07-01 06:23:49 ChiaraRMBP.local werkzeug[34125] INFO 127.0.0.1 - - [01/Jul/2022 06:23:49] "GET /cust000/643594/cancer/4c7d5c70d955875504db72ef8e1abe77?cancer=yes HTTP/1.1"
```

And this PR fixes it nicely! 🥇 

image


image

File: scout/server/blueprints/variant/templates/variant/cancer-variant.html [hunk 1]
```diff
@@ -104,8 +105,8 @@
 
     <div class="row">
       <div class="col-12">{{ overlapping_panel(variant, overlapping_vars, case, institute) }}</div>
-      {% if variant.rank_score_results %}
-        <div class="col-12">{{ rankscore_panel(variant) }}</div>
+      {% if rank_score_results %}
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/templates/variant/cancer-variant.html:108]: 👍🏻

**Review by northwestwitch — CHANGES_REQUESTED** [review]
Good job! The code is fixing the error nicely!

I've noticed that if you hover on the ? in the ranking panel in case page the popovers don't work:

image


It should be like this instead:

image

Since you are already fixing this code could you fix this as well?

You could just copy a few lines from here: https://github.com/Clinical-Genomics/scout/blob/f646360c9ab0e2a98cafc338c0713065535abf8d/scout/server/blueprints/variant/templates/variant/variant.html#L493

Thanks!

**Comment by maehler** [comment]
Sure, I'll take a look at it!

**Comment by northwestwitch** [comment]
I was planning to release a software patch today to fix also another error (#3482). It would be nice to include this one as well, what do you say @maehler ?

**Comment by maehler** [comment]
Sounds good, I'll get on it immediately.

File: scout/server/blueprints/variant/templates/variant/cancer-variant.html [hunk 2]
```diff
@@ -346,6 +346,14 @@ <h6 class="card-header">Variant callers</h6>
       return new bootstrap.Tooltip(tooltipTriggerEl, {container: 'body'})
     })
 
+    var popoverTriggerList = [].slice.call(document.querySelectorAll('[data-bs-toggle="popover"]'))
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/templates/variant/cancer-variant.html:349]: 👍🏻

**Review by northwestwitch — APPROVED** [review]
Nice, now also the tooltips work! 💯

**Comment by northwestwitch** [comment]
I think it's OK to merge now. I don't know if you will be allowed with some tests not completed, otherwise I can do it as admin!

**Comment by northwestwitch** [comment]
Looks like there is a conflict to resolve before merging. It's due to me merging the other PR before this, sorry!

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,9 +4,10 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 About changelog [here](https://keepachangelog.com/en/1.0.0/)
 
-## [Unreleased]
```
Comment (northwestwitch) [inline CHANGELOG.md]: I actually like this. Let's adopt it in the next PRs!

**Pull request merged**